doc/crds.md

```diff
@@ -614,6 +614,29 @@ Notable attributes:
 * **status.state**: Either of `PENDING`, `READY` or `ERROR`. If the state is `ERROR`, the
   attribute `status.errorMessage` contains the reason for the failure.
 
+While it's possible to extend a profile and build it based on another one, it's also
+possible to write a profile from scratch using the `TailoredProfile` construct.
+To do this, remember to set an appropriate title and description. It's very important
+to leave the `extends` field empty for this case. Subsequently, you'll also need
+to indicate to the Compliance Operator what type of scan will this custom profile
+generate:
+
+* Node scan: Scans the Operating System.
+* Platform scan: Scans the OpenShift configuration.
+
+To do this, set the following annotation on the TailoredProfile object:
+
+```
+  compliance.openshift.io/product-type: <Type>
+```
+
+Where the `Platform` type will build a Platform scan, and `Node` will
+build an OS scan. Note that if no `product-type` annotation is given, the
+operator will default to `Platform`. Adding the `-node` suffix to the
+name of the `TailoredProfile` object will have a similar effect as
+adding the `Node` product type annotation, and will generate an Operating
+System scan.
+
```

Comment:
When I use `-node` suffix to make a `TailoredProfile`, it did give correct annotation for tp as `compliance.openshift.io/product-type=Node`, however in `Suites`, it also generated a platform scan besides the node scan, it that a expected behavior? I was using this to test for setting custom variable, and happen to find this.

Here is my tailor profile:
```yaml
apiVersion: compliance.openshift.io/v1alpha1
kind: TailoredProfile
metadata:
  annotations:
    compliance.openshift.io/product-type: Node
  selfLink: >-
    /apis/compliance.openshift.io/v1alpha1/namespaces/openshift-compliance/tailoredprofiles/nist-moderate-modified-node
  resourceVersion: '1108128'
  name: nist-moderate-modified-node
  uid: 3fb95694-70f3-47ce-8f35-2e58a1eae349
  creationTimestamp: '2021-09-02T21:01:25Z'
  generation: 1
  managedFields:
    - apiVersion: compliance.openshift.io/v1alpha1
      fieldsType: FieldsV1
      fieldsV1:
        'f:spec':
          .: {}
          'f:enableRules': {}
          'f:setValues': {}
          'f:title': {}
      manager: Mozilla
      operation: Update
      time: '2021-09-02T21:01:25Z'
    - apiVersion: compliance.openshift.io/v1alpha1
      fieldsType: FieldsV1
      fieldsV1:
        'f:metadata':
          'f:annotations':
            .: {}
            'f:compliance.openshift.io/product-type': {}
          'f:ownerReferences':
            .: {}
            'k:{"uid":"93532d73-68c7-4257-a5c8-b34db4a390c4"}':
              .: {}
              'f:apiVersion': {}
              'f:blockOwnerDeletion': {}
              'f:controller': {}
              'f:kind': {}
              'f:name': {}
              'f:uid': {}
        'f:status':
          .: {}
          'f:id': {}
          'f:outputRef':
            .: {}
            'f:name': {}
            'f:namespace': {}
          'f:state': {}
      manager: compliance-operator
      operation: Update
      time: '2021-09-02T21:01:25Z'
  namespace: openshift-compliance
  ownerReferences:
    - apiVersion: compliance.openshift.io/v1alpha1
      blockOwnerDeletion: true
      controller: true
      kind: ProfileBundle
      name: rhcos4
      uid: 93532d73-68c7-4257-a5c8-b34db4a390c4
spec:
  enableRules:
    - name: rhcos4-service-sshd-disabled
      rationale: test for a custom value
  setValues:
    - name: rhcos4-sshd-idle-timeout-value
      rationale: test for a custom value
      value: '3600'
  title: My modified nist profile with a custom value
status:
  id: xccdf_compliance.openshift.io_profile_nist-moderate-modified-node
  outputRef:
    name: nist-moderate-modified-node-tp
    namespace: openshift-compliance
  state: READY
```

And `ComplianceSuite`: Notes the Not-Applicable is as intended for node scan, since I am not running on rhcos4

```yaml
apiVersion: compliance.openshift.io/v1alpha1
kind: ComplianceSuite
metadata:
  selfLink: >-
    /apis/compliance.openshift.io/v1alpha1/namespaces/openshift-compliance/compliancesuites/nist-moderate-modified-node
  resourceVersion: '1110217'
  name: nist-moderate-modified-node
  uid: bc7afcff-2900-49e2-81e2-ad0a3e793a2a
  creationTimestamp: '2021-09-02T21:01:41Z'
  generation: 1
  managedFields:
    - apiVersion: compliance.openshift.io/v1alpha1
      fieldsType: FieldsV1
      fieldsV1:
        'f:metadata':
          'f:finalizers':
            .: {}
            'v:"suite.finalizers.compliance.openshift.io"': {}
          'f:ownerReferences':
            .: {}
            'k:{"uid":"d3773f06-2d61-4638-a269-682a0549145d"}':
              .: {}
              'f:apiVersion': {}
              'f:blockOwnerDeletion': {}
              'f:controller': {}
              'f:kind': {}
              'f:name': {}
              'f:uid': {}
        'f:spec':
          .: {}
          'f:scans': {}
          'f:schedule': {}
        'f:status':
          .: {}
          'f:conditions': {}
          'f:phase': {}
          'f:result': {}
          'f:scanStatuses': {}
      manager: compliance-operator
      operation: Update
      time: '2021-09-02T21:02:05Z'
  namespace: openshift-compliance
  ownerReferences:
    - apiVersion: compliance.openshift.io/v1alpha1
      blockOwnerDeletion: true
      controller: true
      kind: ScanSettingBinding
      name: nist-moderate-modified-node
      uid: d3773f06-2d61-4638-a269-682a0549145d
  finalizers:
    - suite.finalizers.compliance.openshift.io
spec:
  scans:
    - content: ssg-ocp4-ds.xml
      contentImage: 'quay.io/complianceascode/ocp4:latest'
      name: ocp4-moderate
      profile: xccdf_org.ssgproject.content_profile_moderate
      rawResultStorage:
        pvAccessModes:
          - ReadWriteOnce
        rotation: 3
        size: 1Gi
      scanTolerations:
        - effect: NoSchedule
          key: node-role.kubernetes.io/master
          operator: Exists
      scanType: Platform
    - nodeSelector:
        node-role.kubernetes.io/worker: ''
      contentImage: 'quay.io/complianceascode/ocp4:latest'
      profile: xccdf_compliance.openshift.io_profile_nist-moderate-modified-node
      name: nist-moderate-modified-node-worker
      tailoringConfigMap:
        name: nist-moderate-modified-node-tp
      rawResultStorage:
        pvAccessModes:
          - ReadWriteOnce
        rotation: 3
        size: 1Gi
      scanType: Node
      content: ssg-rhcos4-ds.xml
      scanTolerations:
        - effect: NoSchedule
          key: node-role.kubernetes.io/master
          operator: Exists
    - nodeSelector:
        node-role.kubernetes.io/master: ''
      contentImage: 'quay.io/complianceascode/ocp4:latest'
      profile: xccdf_compliance.openshift.io_profile_nist-moderate-modified-node
      name: nist-moderate-modified-node-master
      tailoringConfigMap:
        name: nist-moderate-modified-node-tp
      rawResultStorage:
        pvAccessModes:
          - ReadWriteOnce
        rotation: 3
        size: 1Gi
      scanType: Node
      content: ssg-rhcos4-ds.xml
      scanTolerations:
        - effect: NoSchedule
          key: node-role.kubernetes.io/master
          operator: Exists
  schedule: 0 1 * * *
status:
  conditions:
    - lastTransitionTime: '2021-09-02T21:04:58Z'
      message: Compliance suite run is done running the scans
      reason: NotRunning
      status: 'False'
      type: Processing
    - lastTransitionTime: '2021-09-02T21:04:58Z'
      message: Compliance suite run is done and has results
      reason: Done
      status: 'True'
      type: Ready
  phase: DONE
  result: NON-COMPLIANT
  scanStatuses:
    - name: nist-moderate-modified-node-worker
      phase: DONE
      result: NOT-APPLICABLE
      resultsStorage:
        name: nist-moderate-modified-node-worker
        namespace: openshift-compliance
    - name: ocp4-moderate
      phase: DONE
      result: NON-COMPLIANT
      resultsStorage:
        name: ocp4-moderate
        namespace: openshift-compliance
      warnings: >-
        could not fetch
        /apis/logging.openshift.io/v1/namespaces/openshift-logging/clusterlogforwarders/instance:
        the server could not find the requested resource

        could not fetch
        /apis/logging.openshift.io/v1/namespaces/openshift-logging/clusterlogforwarders/instance:
        the server could not find the requested resource

        could not fetch
        /apis/console.openshift.io/v1/consolenotifications/classification-banner:
        consolenotifications.console.openshift.io "classification-banner" not
        found

        could not fetch
        /apis/fileintegrity.openshift.io/v1alpha1/fileintegrities?limit=5: the
        server could not find the requested resource

        could not fetch
        /apis/machineconfiguration.openshift.io/v1/machineconfigs/99-master-fips:
        machineconfigs.machineconfiguration.openshift.io "99-master-fips" not
        found

        could not fetch /api/v1/namespaces/openshift/configmaps/motd: configmaps
        "motd" not found
    - name: nist-moderate-modified-node-master
      phase: DONE
      result: NOT-APPLICABLE
      resultsStorage:
        name: nist-moderate-modified-node-master
        namespace: openshift-compliance
```

Reply:
this is definitely not expected behavior, let me try it out.

Reply:
I was unable to replicate this. perhaps there was a suite with a similar name on the system already? Might want to try this out on a clean cluster.

Reply:
Thanks for testing it out. I think I was previous using a tailor profile extending the moderate profile, then I deleted that tailor profile and created a new one above, after that I deleted the suite, maybe it is with the last tailor profile, I didn't fully removed them from the cluster.

The code /lgtm.

Reply:
Sorry about the confusion, the platform scan was from my scansettingbindings!